### PR TITLE
Feature/task5 scss

### PR DESCRIPTION
**Pull request by KirillPrikhodko**

### CODE REVIEW

File: Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__flag/flex-container__flag-text/flex-container__flag-text.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.flex-container__flag-text {
+    position: absolute;
+    left: 9px;
+    height: 82px;
+    width: 70px;
+    font-size: $regular-font;
+    font-family: 'Raleway', sans-serif;
+    color: $white-color;
+    z-index: 200;
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__flag/flex-container__flag-text/flex-container__flag-text.scss]: It seems unnecessary.

File: Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer--desktop-only.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.flex-container__offer--desktop-only {
+    @include flex-container__offer;
+    order: 3;
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer--desktop-only.scss]: Why?

File: Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer-price-before-discount/flex-container__offer-price-before-discount.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+.flex-container__offer-text .flex-container__offer-price-before-discount {
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer-price-before-discount/flex-container__offer-price-before-discount.scss]: I can't find this case in the markup. Is it needed?

File: Task3.Rwd/src/scss/mixins/mixins.scss [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        left: -10px;
+    }
+}
+
+@mixin raleway-dark-blue {
+    font-size: $regular-font;
+    font-family: "Raleway", sans-serif;
+    color: $dark-blue-color;
+    margin-left: 5px;
+    z-index: 15;
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/mixins/mixins.scss]: Lines 37 and 38 seem very weird for a mixin that seemingly deifnes a text of a certain font and color.

File: Task3.Rwd/src/scss/mixins/mixins.scss [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+@mixin followed-by-black-arrow-down {
+    &::after {
+        position: relative;
+        left: 5px;
+        top: -2px;
+        content: url(assets/images/svg/black-arrow-down.svg);
+    }
+}
+
+@mixin flex-container__offer {
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/mixins/mixins.scss]: A mixin shouldn't be an element.

File: Task3.Rwd/src/scss/mixins/mixins.scss [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    
+    @media (min-width: $desktop-size) {
+            width: 555px;
+            margin: 11px 15px;
+            height: auto;
+    }
+}
+
+@mixin dark-blue-link {
+    &:link, &:active, &:visited {
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/mixins/mixins.scss]: It's semantically wrong for all these states to have the same color/decoration.

File: Task3.Rwd/src/scss/mixins/mixins.scss [hunk 7]
```diff
@@ -0,0 +1,100 @@
+@mixin header-panel__child {
+    vertical-align: middle;
+    margin-left: 35px;
+}
+
+@mixin button {
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/mixins/mixins.scss]: Consider making it a block.

File: Task3.Rwd/src/scss/mixins/mixins.scss [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    position: relative;
+
+    &::before {
+        position: absolute;
+        top: 3px;
+        left: -10px;
+    }
+}
+
+@mixin raleway-dark-blue {
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/mixins/mixins.scss]: It's very bad to name mixins/variables/classes by their appearance. The names must be semantic.

File: Task3.Rwd/src/scss/variables/variables.scss [hunk 9]
```diff
@@ -0,0 +1,17 @@
+// device sizes
+$mobile-size: 320px;
+$tablet-size: 768px;
+$desktop-size: 1400px;
+
+// font sizes
+$small-font: 10px;
+$regular-font: 14px;
+$bigger-font: 18px;
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/variables/variables.scss]: Please come come with meaningful semantic names for font sizes.

File: Task3.Rwd/src/scss/variables/variables.scss [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+$big-font: 30px;
+
+//colors
+$dark-blue-color: #0d3351;
+$white-color: #ffffff;
+$gray-color: #f3f3f3;
+$light-blue-color: #00a2dd;
+$super-light-blue-color: #33c7fd;
```
Comment (ahryhoryeu) [inline Task3.Rwd/src/scss/variables/variables.scss]: AS I mentioned in another comment, it's very bad to name variables after their appearances. Color should be either named by their meaning/usage (like `default-text-color`, `highlighted-text-color`) or, even better, just by numbers, i.e. `color-1`, `color-2` and so on.

File: Task3.Rwd/webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+      chunkFilename: '[id].css',
+      ignoreOrder: false, // Enable to remove warnings about conflicting order
+    }),
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.(gif|png|jpe?g|svg)$/i,
+        use: [
+          'file-loader?name=img/[name].[ext]',
```
Comment (ahryhoryeu) [inline Task3.Rwd/webpack.config.js]: It's a deprecated and non-maintainable syntax to put loader parameters in a string value.

File: Task3.Rwd/webpack.config.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+              
+              hmr: process.env.NODE_ENV === 'development',
+            },
+          },
+          'css-loader',
+          'sass-loader',
+        ],
+      },
+    ],
+  },
```
Comment (ahryhoryeu) [inline Task3.Rwd/webpack.config.js]: Please remove dangling commas.

File: Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__flag/flex-container__flag-text/flex-container__flag-text.scss (hunk 1 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__flag/flex-container__flag-text/flex-container__flag-text.scss]: fixed

File: Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer--desktop-only.scss (hunk 2 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer--desktop-only.scss]: the order of flex elements (as stated in .psd mockup) is different on desktop/tablet/mobile devices, so I came up with this solution.

File: Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer-price-before-discount/flex-container__offer-price-before-discount.scss (hunk 3 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/main/flex-container-wrapper/flex-container/flex-container__offer/flex-container__offer-price-before-discount/flex-container__offer-price-before-discount.scss]: removed

File: Task3.Rwd/src/scss/mixins/mixins.scss (hunk 4 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/mixins/mixins.scss]: removed

File: Task3.Rwd/src/scss/mixins/mixins.scss (hunk 5 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/mixins/mixins.scss]: fixed

File: Task3.Rwd/src/scss/mixins/mixins.scss (hunk 8 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/mixins/mixins.scss]: fixed

File: Task3.Rwd/src/scss/variables/variables.scss (hunk 9 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/variables/variables.scss]: fixed

File: Task3.Rwd/src/scss/variables/variables.scss (hunk 10 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/variables/variables.scss]: fixed

File: Task3.Rwd/webpack.config.js (hunk 11 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/webpack.config.js]: fixed

File: Task3.Rwd/webpack.config.js (hunk 12 above)
Comment (KirillPrikhodko) [reply Task3.Rwd/webpack.config.js]: fixed

**Comment by KirillPrikhodko** [comment]
> 3. Please make use of `stylelint` library. At the moment there are plenty of cases of broken/inconsistent formatting in your styles.

I think I made it work but I am not sure if I configured everything properly. For instance, I have troubles configuring it to work with scss-specific syntax (nested rules, mixins and variables).

> 5. There are several places where thing could've been done with CSS rather than images/additional elements, like flags on offers (an element with some before/after pseudo-elements, and also a separate block), "followed by black arrow down" - a simple `::after` pseudo-element, the same for "arrow-right" image.

For some reason, a triangle pointing right does not get created correctly (I left a respective comment at line 27 src/scss/blocks/offer-text/offer-text.scss
As to the flags, I tried to do it but I do not know how to make them transparent (maybe we should discuss it offline).

File: Task3.Rwd/src/scss/blocks/choose-cruise-button/choose-cruise-button.scss [hunk 13]
```diff
@@ -0,0 +1,9 @@
+@import "../../../scss/mixins/mixins.scss";
```
[the remark this answers is not in the document]
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/blocks/choose-cruise-button/choose-cruise-button.scss]: fixed

File: Task3.Rwd/src/scss/blocks/flag/flag.scss [hunk 14]
```diff
@@ -0,0 +1,17 @@
+.flag__image {
```
[the remark this answers is not in the document]
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/blocks/flag/flag.scss]: fixed

File: Task3.Rwd/src/scss/blocks/footer/footer.scss [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+    text-align: left;
+    padding: 20px 5px 30px 15px;
+  }
+
+  @media (min-width: $desktop-size) {
+    padding: 20px 380px;
+  }
+}
+
+.footer__social-links {
```
[the remark this answers is not in the document]
Comment (KirillPrikhodko) [reply Task3.Rwd/src/scss/blocks/footer/footer.scss]: fixed

**Pull request merged**